tests/Unit/DataStructures/Test_VectorImplHelper.hpp

```diff
@@ -177,5 +183,361 @@ auto remove_nth(const std::tuple<ValueTypes...>& tup) noexcept {
+                                             make_not_null(&dist)),
+       v2 = make_with_random_values<ValType>(make_not_null(&gen),
+                                             make_not_null(&dist));
+  fill_with_random_values(make_not_null(&v1), make_not_null(&gen),
+                          make_not_null(&dist));
+  fill_with_random_values(make_not_null(&v2), make_not_null(&gen),
+                          make_not_null(&dist));
+  ValType v1p2 = v1 + v2, v1m2 = v1 - v2;
+
+  VectorType m0{s1, v1}, m1{s1, v2};
+  SECTION("Check move assignment and use after move"){
+    VectorType a{s1, 0};
+    fill_with_random_values(make_not_null(&a), make_not_null(&gen),
+                            make_not_null(&dist));
+    VectorType b{};
+    b = std::move(a);
+    b = m0 + m1;
+    check_vectors(b, VectorType{s1, v1p2});
+    // clang-tidy: use after move (intentional here)
+    CHECK(a.size() == 0);  // NOLINT
+    CHECK(a.is_owning());
+    a = m0 - m1;
+    check_vectors(a, VectorType{s1, v1m2});
+    check_vectors(b, VectorType{s1, v1p2});
+  }
+
+  SECTION("Check move assignment and value of target"){
+    auto v3 = make_with_random_values<ValType>(make_not_null(&gen),
+                                               make_not_null(&dist));
+    VectorType a{s1, v3};
+    VectorType b{};
+    b = std::move(a);
+    a = m0 + m1;
+    check_vectors(b, VectorType{s1, v3});
+    check_vectors(a, VectorType{s1, v1p2});
+  }
+
+  SECTION("Check move constructor and use after move"){
+    VectorType a{s1, 0};
+    fill_with_random_values(make_not_null(&a), make_not_null(&gen),
+                            make_not_null(&dist));
+    VectorType b{std::move(a)};
+    b = m0 + m1;
+    CHECK(b.size() == s1);
+    check_vectors(b, VectorType{s1, v1p2});
+    // clang-tidy: use after move (intentional here)
+    CHECK(a.size() == 0);  // NOLINT
+    CHECK(a.is_owning());
+    a = m0 - m1;
+    check_vectors(a, VectorType{s1, v1m2});
+    check_vectors(b, VectorType{s1, v1p2});
+  }
+
+  SECTION("Check move constructor and value of target"){
+    auto v3 = make_with_random_values<ValType>(make_not_null(&gen),
+                                               make_not_null(&dist));
+    VectorType a{s1, v3};
+    VectorType b{std::move(a)};
+    a = m0 + m1;
+    check_vectors(b, VectorType{s1, v3});
+    check_vectors(a, VectorType{s1, v1p2});
+  }
+  SECTION("Check set_data_ref"){
+    VectorType ref1{s1, v1};
+    VectorType ref2{s1, v2};
+    ref1.set_data_ref(&ref2);
+    ref1 = ref1 + m0;
+    check_vectors(ref2, VectorType{s1,v1p2});
+    check_vectors(ref1, VectorType{s1,v1p2});
+  }
+}
 }  // namespace VectorImpl
 }  // namespace TestHelpers
+
+/// \ingroup TestingFrameworkGroup
+/// \brief a macro to easily generate the basic construction and serialization
+/// test for a vector type `TYPE` and a value type `VALTYPE
+#define DEFINE_BASIC_VECTOR_TEST_CASES(TYPE, VALTYPE)                         \
```

Comment:
make a single test case for efficiency, put each function in its own section for granularity.